File: Excel/614 Challenge.py
import pandas as pd

path = "614 Consecutive Numbers Count.xlsx"

input = pd.read_excel(path, usecols="A", nrows=20)
test = pd.read_excel(path, usecols="B", nrows=20).fillna("")

def consecutive_id(series):
    return (series != series.shift()).cumsum()

input['group'] = consecutive_id(input.iloc[:, 0])
input['Answer Expected'] = input.groupby('group').transform(lambda x: x.size if x.size >= 2 else "")
result = input[['Answer Expected']]

print(result.equals(test)) #True

File: Excel/600-699/614/614 Challenge.R
library(tidyverse)
library(readxl)

path = "Excel/614 Consecutive Numbers Count.xlsx"
input = read_excel(path, range = "A1:A20")
test  = read_excel(path, range = "B1:B20")

result = input %>%
  mutate(group = consecutive_id(Problem)) %>%
  mutate(`Answer Expected` = ifelse(n() >= 2, n(), NA), .by = group) %>%
  select(`Answer Expected`)

all.equal(result, test, check.attributes = FALSE)
#> [1] TRUE